mempool/clist_mempool.go

```diff
@@ -45,6 +45,11 @@ type CListMempool struct {
 
 	// Keeps track of the rechecking process.
 	recheck *recheck
+	// It is true while TXs are being rechecked
+	rechecking atomic.Bool
```

Comment:
This variable is redundant. It's equivalent to `recheck.done()`.

Reply:
After suggesting the above I realized that they're not equivalent because `recheck.Done()` is equal to `recheck.cursor == nil`, which is not thread safe. We may have a scenario where recheck is updating ``recheck.cursor` and `PreUpdate` is reading it. Still, I don't see any race condition on the unit tests, even after fixing them (see comment below). Probably we need new tests.

Reply:
How about leaving it as it is now (even if it suboptimal in terms of variables), so we're confident we can backport it, and then, in a furhter PR, we improve this on `main` and possibly backport it to `v1.x` (or not).
WDYT?

Reply:
I see this has been changed. Shall we revert it?

Reply:
That's an option. But it could be confusing because we'll have two ways to check if rechecking is happening. Another option is #3333.